packages/vscode-processutils/src/utils/spawnStreamAsync.ts

```diff
@@ -82,14 +83,30 @@ export async function spawnStreamAsync(
         throw new CancellationError('Command cancelled', cancellationToken);
     }
 
-    const safeCommand = !!options.allowUnsafeExecutablePath ? command : getSafeExecPath(command);
+    let finalCommand: string;
+    if (!!options.allowUnsafeExecutablePath) {
+        // If allowUnsafeExecutablePath is true, we assume the command is a full command line
+        // and we do not apply any quoting or checks.
+        finalCommand = command;
+    } else {
+        // Otherwise, we do some checks and quoting.
+        const safeCommand = getSafeExecPath(command);
+        const quotedSafeCommand = `"${safeCommand}"`;
+        finalCommand = !!shell ? quotedSafeCommand : safeCommand;
```

Comment:
This unfortunately adds quotes on Unix when they aren't needed. Should we address that or just live with it?